src/Shared/ObjectMethodExecutor/ObjectMethodExecutorFSharpSupport.cs

```diff
@@ -51,33 +74,110 @@ internal static class ObjectMethodExecutorFSharpSupport
         var awaiterResultType = possibleFSharpAsyncType.GetGenericArguments().Single();
         awaitableType = typeof(Task<>).MakeGenericType(awaiterResultType);
 
-        // coerceToAwaitableExpression = (object fsharpAsync) =>
+        // coerceToAwaitableExpression = (FSharpAsync<TResult> fsharpAsync) =>
         // {
-        //     return (object)FSharpAsync.StartAsTask<TResult>(
-        //         (Microsoft.FSharp.Control.FSharpAsync<TResult>)fsharpAsync,
+        //     return FSharpAsync.StartAsTask<TResult>(
+        //         fsharpAsync,
         //         FSharpOption<TaskCreationOptions>.None,
         //         FSharpOption<CancellationToken>.None);
         // };
+
         var startAsTaskClosedMethod = _fsharpAsyncStartAsTaskGenericMethod
             .MakeGenericMethod(awaiterResultType);
-        var coerceToAwaitableParam = Expression.Parameter(typeof(object));
-        coerceToAwaitableExpression = Expression.Lambda(
-            Expression.Convert(
-                Expression.Call(
-                    startAsTaskClosedMethod,
-                    Expression.Convert(coerceToAwaitableParam, possibleFSharpAsyncType),
-                    Expression.MakeMemberAccess(null, _fsharpOptionOfTaskCreationOptionsNoneProperty),
-                    Expression.MakeMemberAccess(null, _fsharpOptionOfCancellationTokenNoneProperty)),
-                typeof(object)),
-            coerceToAwaitableParam);
+
+        var coerceToAwaitableParam = Expression.Parameter(possibleFSharpAsyncType);
+
+        var startAsTaskCall =
+            Expression.Call(
+                method: startAsTaskClosedMethod,
+                arg0: coerceToAwaitableParam,
+                arg1: Expression.MakeMemberAccess(null, _fsharpOptionOfTaskCreationOptionsNoneProperty),
+                arg2: Expression.MakeMemberAccess(null, _fsharpOptionOfCancellationTokenNoneProperty));
```

Comment:
Would it make sense to pass in the request’s cancellation token? Otherwise the user would have to set it themselves making the changes here not useful.

Reply:
Hmm. I think it would be odd to have an explicitly-declared but implicitly-used[^1] cancellation token parameter. The IDE would show the `cancellationToken` parameter as being unused in the following examples—but with your suggestion it actually _would_ be used implicitly at runtime:

##### *Minimal API endpoint with explicitly-declared, implicitly-used cancellation token

```fsharp
get "/" (fun (db : IDb) (cancellationToken : CancellationToken) -> async {
    let! results = db.Get ()
    return results
})
```

##### *Controller method with explicitly-declared, implicitly-used cancellation token

```fsharp
[<HttpGet("")>]
member _.Get (cancellationToken : CancellationToken) =
    async {
        let! results = db.Get ()
        return results
    }
```

Since the user must already explicitly declare a cancellation token parameter if they desire cancellability, I think it makes sense that they must also pass the token explicitly if they wish to use it, e.g.:

##### Minimal API endpoint with explicit token passing

```fsharp
get "/" (fun (db : IDb) (cancellationToken : CancellationToken) -> Async.StartAsTask (async {
    let! results = db.Get ()
    return results
}, cancellationToken=cancellationToken))
```

##### Controller method with explicit token passing

```fsharp
[<HttpGet("")>]
member _.Get (cancellationToken : CancellationToken) =
    Async.StartAsTask (async {
        let! results = db.Get ()
        return results
    }, cancellationToken=cancellationToken)
```

[^1]: The framework does automatically (and implicitly) bind the cancellation token parameter to `HttpContext.RequestAborted` regardless, but this has no effect on the execution of any asynchronous operations (whether `Async<_>` or `Task<_>`) that are invoked in the action/endpoint.